src/diagnostics/MOM_wave_structure.F90

```diff
@@ -195,7 +197,13 @@ subroutine wave_structure(h, tv, G, GV, US, cn, ModeNum, freq, CS, En, full_halo
   Pi = (4.0*atan(1.0))
 
   S => tv%S ; T => tv%T
-  g_Rho0 = US%L_T_to_m_s**2 * GV%g_Earth / GV%Rho0
+  g_Rho0 = GV%g_Earth / GV%Rho0
+
+  if (CS%debug) call chksum0(g_Rho0, "g/rho0 in wave struct", &
+                             scale=US%Z_to_m*(US%s_to_T**2)*US%kg_m3_to_R)
```

Comment:
I suspect that the scaling here is not quite right, and that it requires an additional factor of US%L_to_Z**2.  See the documentation of the rescaled units for GV%g_Earth in MOM_vertical_grid.F90.  Also, once this is working we may want to comment out this checksum for a value that is the same for each call, or only do the checksum the first time this routine is called.